lib/internal/modules/esm/resolve.js

```diff
@@ -255,18 +256,12 @@ function finalizeResolution(resolved, base) {
       resolved.pathname, fileURLToPath(base), 'module');
   }
 
-  let path;
-  try {
-    path = fileURLToPath(resolved);
-  } catch (err) {
-    if (err.code === 'ERR_INVALID_FILE_URL_PATH' &&
-        err.message.includes('must not include encoded "/" characters')) {
-      throw new ERR_INVALID_MODULE_SPECIFIER(
-        resolved.pathname, 'must not include encoded "/" characters',
-        fileURLToPath(base));
-    }
-    throw err;
-  }
+  if (encodedSepRegEx.test(resolved.pathname))
```

Comment:
```suggestion
  if (RegExpPrototypeTest(encodedSepRegEx, resolved.pathname))
```

Reply:
(sorry i couldn't add the import; i can't comment on lines that are too far from diffs, so i can't suggest that high up in the file)